tests/ipfwd/test_nhop_count.py

```diff
@@ -365,9 +371,18 @@ def test_nhop(request, duthost, tbinfo):
     # verify the test used up all the NHOP group resources
     # skip this check on Mellanox as ASIC resources are shared
     if not is_mellanox_device(duthost):
-        pytest_assert(
-            crm_after["available"] == 0,
-            "Unused NHOP group resource: {}, used:{}".format(
-                crm_after["available"], crm_after["used"]
+        if is_cisco_device(duthost):
```

Comment:
Make is_cisco_device and is_mellanox_device  in the same column, avoiding unnecessary condition wrap.

Reply:
there are 3 conditions here.
A1. not is_mellanox_device, is_cisco_device is <checking to 92%>
A2. not is_mellanox_device and not is_cisco_device, others are <checking to 100%>
B. is_mellanox_device <skip all checking>

Reply:
yes, I mean don't warp is_cisco_device into branch "if not is_mellanox_device(duthost)", that looks complicate.
something like this:
if cisco:
     
elif mellanox:

else

Reply:
thanks for the advise. I have updated the codes as above format.
let me know anything I can improve my codes.